client/pages/index.tsx

```diff
@@ -22,13 +23,17 @@ const pageStates = {
   RESULTS: 'RESULTS', // displays results
   API_ERROR: 'API_ERROR', // API error page
   ERROR: 'ERROR', // other errors
+  ERROR_POPUP: 'ERROR_POPUP', // displays error message popup
 };
 
 const Home = () => {
   const { takePhoto } = useCamera();
 
   const [pageState, setPageState] = useState(pageStates.MAIN);
   const [productInfo, setProductInfo] = useState<ProductData[]>([]);
+  const [errorMessage, setErrorMessage] = useState<string | null>(null); // error message
+  const [showErrorPopup, setShowErrorPopup] = useState(false); // show error popup flag
```

Comment:
maybe try using `pageState` and `setPageState` to control the `showErrorPopup` instead of having another state?
it might keep things more "tidy"? let me know what you think!